syntax = "proto3";

package testrpc;

import "google/protobuf/empty.proto";

service TestgRPC {

    rpc get_json(google.protobuf.Empty) returns (ListOperators) {}	    	
	
    rpc get_result(google.protobuf.Empty) returns (Operator) {}

    rpc record_result(stream Variables) returns (OperatorsSummary) {}

    rpc result_chat(stream SummNote) returns (stream SummNote) {}

}


message Variables {
    // value for operators calculate
    int32 value = 1;
    int32 value_2 = 2;
}

message ListOperators {
   repeated Operator operators = 1;
}

message Operator {
    // operator name
    string name = 1;

    //result summary
    Response response_ = 2;

    Variables items = 3;
}

message Response {
    int32 result_ = 1;
}

message SummNote {
    // kind of operator
    Variables type_operator = 1;

    // message to sent
    string message = 2;
}

message OperatorsSummary {
    // how much value for summ sent
    int32 variables_count = 1;
    // how much operators sent
    int32 feature_count = 2;

    int32 elapsed_time = 3;
}